observer/circularQueueNodesProvider.go

```diff
@@ -51,11 +51,11 @@ func (cqnp *circularQueueNodesProvider) GetNodesByShardId(shardId uint32) ([]*da
 }
 
 // GetAllNodes will return a slice containing all observers
-func (cqnp *circularQueueNodesProvider) GetAllNodes() ([]*data.NodeData, error) {
+func (cqnp *circularQueueNodesProvider) GetAllNodes(dataAvailability data.ObserverDataAvailabilityType) ([]*data.NodeData, error) {
 	cqnp.mutNodes.Lock()
```

Comment:
RLock ?

Reply:
since I also perform write operations, I would leave it as it is